include/zephyr/net/trickle.h

```diff
@@ -51,15 +51,15 @@ typedef void (*net_trickle_cb_t)(struct net_trickle *trickle,
  * only via the Trickle API.
```

Comment:
Please remove the `subsys/` from the commit subject, `net: trickle: ...` should be fine there.